corehq/ex-submodules/casexml/apps/case/cleanup.py

```diff
@@ -73,6 +72,12 @@ def safe_hard_delete(case):
 
     This is used primarily for cleaning up system cases/actions (e.g. the location delegate case).
     """
+    if not settings.UNIT_TESTING:
+        from corehq.apps.commtrack.const import USER_LOCATION_OWNER_MAP_TYPE
+        whitelist_case_types = [USER_LOCATION_OWNER_MAP_TYPE]
+        if case.type not in whitelist_case_types:
```

Comment:
Don't plan to add more types but this just felt like the intention was more
clear. Suppose a comment would also be fine.

On 20 November 2015 at 16:47, Daniel Miller <notifications@github.com>
wrote:

> In corehq/ex-submodules/casexml/apps/case/cleanup.py
> <https://github.com/dimagi/commcare-hq/pull/9198#discussion_r45475735>:
>
> > @@ -73,6 +72,12 @@ def safe_hard_delete(case):
> >
> >      This is used primarily for cleaning up system cases/actions (e.g. the location delegate case).
> >      """
> > +    if not settings.UNIT_TESTING:
> > +        from corehq.apps.commtrack.const import USER_LOCATION_OWNER_MAP_TYPE
> > +        whitelist_case_types = [USER_LOCATION_OWNER_MAP_TYPE]
> > +        if case.type not in whitelist_case_types:
>
> Is there a plan to add more types to this list? If not, how about case.type
> == USER_LOCATION_OWNER_MAP_TYPE?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/9198/files#r45475735>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa